daemon/build.go

```diff
@@ -165,7 +165,8 @@ func (daemon *Daemon) pullForBuilder(ctx context.Context, name string, authConfi
 		pullRegistryAuth = &resolvedConfig
 	}
 
-	if err := daemon.pullImageWithReference(ctx, ref, platform, nil, pullRegistryAuth, output); err != nil {
+	_, err = daemon.pullImageWithReference(ctx, ref, platform, nil, pullRegistryAuth, output)
+	if err != nil {
```

Comment:
Was there a reason you changed the scope of the `err` here, instead of:

```go
if _, err := daemon.pullImageWithReference(ctx, ref, platform, nil, pullRegistryAuth, output); err != nil {
```